metro.go

```diff
@@ -93,16 +106,19 @@ func initLogging(to_file bool, level string) {
 	logger, err := log.LoggerFromConfigAsBytes(logConfig)
 	if err != nil {
 		log.Criticalf("Unable to initiate logger: %s", err)
-		os.Exit(1)
+		panic(Exit{1})
 	}
 	log.ReplaceLogger(logger)
+	return logger
 
 }
 
 func main() {
+	defer handleExit()
 	flag.Parse()
 
-	initLogging(true, "warning")
+	logger := initLogging(true, "warning")
+	defer logger.Flush()
```

Comment:
From https://github.com/cihub/seelog/wiki/Quick-start#defer-block-and-flushing, it looks like you only need to `defer log.Flush()` at the beginning of the `main` function, before any `seelog` construct is used.

Is there a reason you did it differently here?